### ISSUE TITLE
BuildEndpointDefinitions fails when adding validators

### ISSUE THREAD

**Issue by Allann** [dj-nitehawk/FastEndpoints#128]
I have two Validators that both validator a string type (CallsignValidator and SignInNameValidator), both are used in my only endpoint (ShiftStart) and when the startup logic gets to the BuildEndpointDefinitions call where it's adding IValidators to the valDict it uses the tRequest as the key.  This key in both cases happens to be System.String which then throws a ArgumentException
```
System.ArgumentException: An item with the same key has already been added. Key: System.String
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at FastEndpoints.EndpointData.BuildEndpointDefinitions(IServiceCollection services, EndpointDiscoveryOptions options, ConfigurationManager config)
   at FastEndpoints.EndpointData.<>c__DisplayClass6_0.<.ctor>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at FastEndpoints.EndpointData..ctor(IServiceCollection services, EndpointDiscoveryOptions options, ConfigurationManager config)
   at FastEndpoints.MainExtensions.AddFastEndpoints(IServiceCollection services, EndpointDiscoveryOptions options, ConfigurationManager config)
   at Program.<Main>$(String[] args) in C:\Users\allan\source\Nhvr\Nhvr.Rcms.Officer\src\Nhvr.Rcms.Shift.Api\Program.cs:line 4
```

I have included my validators for completeness here:
``` cs
public sealed class CallSignValidator : Validator<string>
{
    public CallSignValidator()
    {
        RuleFor(c => c)
            .NotNull()
            .NotEmpty()
            .MaximumLength(16)
            .Matches("^[A-Za-z0-9-_.~]{1,16}$").WithMessage("CallSign contains invalid characters, only use alpha, numeric and -_.~ are valid.")
            .Must(c => !c.StartsWith('.')).WithMessage("CallSign should not start with a '.'");
    }
}
```
``` cs
public class SignInNameValidator : Validator<string>
{
    public SignInNameValidator()
    {
        RuleFor(c => c)
            .NotNull()
            .NotEmpty()
            .MaximumLength(254)
            .Matches("[^@ \t\r\n]+@[^@ \t\r\n]+\\.[^@ \t\r\n]+").WithMessage($"{nameof(SignInName)} should be an email address.");
    }
}
```
I tried this this with the latest full release and the current beta (4.3.0-beta2).

**Comment by dj-nitehawk** [comment]
afaik this is not how fluentvalidation rules are meant to be used.

here's what the author of fluentvalidations has to say about it:
https://github.com/FluentValidation/FluentValidation/issues/184

here's the recommended solution to write re-usable validators for scalar/primitive types (using extension methods):
https://github.com/FluentValidation/FluentValidation/issues/184#issuecomment-197952324

basically, in fastendpoints, you need to write a validator per request dto:
`Validator<MyRequestA>`
`Validator<MyRequestB>`
 etc.

**Comment by Allann** [comment]
Wow, I've been doing it wrong for years and you are the first to pull me up on it.  Thanks for the link and explanation.
